### PR TITLE
Configure DNS servers in machine allocation optionally

### PR DESCRIPTION
**Pull request by simcod**

### CODE REVIEW

File: cmd/metal-api/internal/service/v1/machine.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ type MachineAllocation struct {
 	VPN              *MachineVPN               `json:"vpn" description:"vpn connection info for machine" optional:"true"`
 	AllocationUUID   string                    `json:"allocationuuid" description:"a unique identifier for this machine allocation, can be used to distinguish between machine allocations over time."`
 	FirewallRules    *FirewallRules            `json:"firewall_rules,omitempty" description:"a set of firewall rules to apply" optional:"true"`
+	DNSServers       []string                  `json:"dns_servers,omitempty" description:"the dns servers used for the machine" optional:"true"`
```
Comment (majst01) [inline cmd/metal-api/internal/service/v1/machine.go:48]: you must also put the DNSService into the response

File: cmd/metal-api/internal/service/machine-service.go [hunk 2]
```diff
@@ -1237,6 +1239,7 @@ func allocateMachine(ctx context.Context, logger *slog.Logger, ds *datastore.Ret
 		VPN:             allocationSpec.VPN,
 		FirewallRules:   firewallRules,
 		UUID:            uuid.New().String(),
+		DNSServers:      allocationSpec.DNSServers,
```
Comment (majst01) [inline cmd/metal-api/internal/service/machine-service.go:1242]: some sort of validation is required like:
walk through all elements
if the element is a ip, validate with netip.ParseAddr()
else it must be a rfc conformant hostname

Comment (majst01) [reply cmd/metal-api/internal/service/machine-service.go:1242]: This ensures a breaking OS installation

**Comment by simcod** [comment]
Will be handled in a PR with dns and ntp combined.

**Pull request closed without merging**